api/v1beta2/bucket_types.go

```diff
@@ -78,11 +78,15 @@ type BucketSpec struct {
 	SecretRef *meta.LocalObjectReference `json:"secretRef,omitempty"`
 
 	// Interval at which to check the Endpoint for updates.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
 	// +required
 	Interval metav1.Duration `json:"interval"`
 
 	// Timeout for fetch operations, defaults to 60s.
 	// +kubebuilder:default="60s"
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m))+$"
```

Comment:
Did not include `h` for timeouts (nor did I include `ns`, `us`, `µs`) to stimulate the user to keep their bounds realistic.